### PR TITLE
Add source map

### PR DESCRIPTION
**Pull request by Johann-S**
Add sourcesmaps related to this issue : https://github.com/twbs/ratchet/issues/410

### CODE REVIEW

**Comment by Johann-S** [comment]
Any feedbacks about this @cvrebert ?

**Comment by cvrebert** [comment]
Seems fine to me, although I'm not part of Team Ratchet.

**Comment by Johann-S** [comment]
Thanks for your feedbacks.
Sorry to ask that but which member of the twbs team are part of Team Ratchet ? And Which one can I CC for my pull request(s) ?

**Comment by XhmikosR** [comment]
@Johann-S: please fetch and rebase.

**Comment by Johann-S** [comment]
Sorry I'm not sure about command to use to fetch and rebase. 
Something like this : 
``git fetch upstream master`` and
``git rebase -i HEAD~2``

**Comment by XhmikosR** [comment]
```
git fetch upstream master
git rebase -i upstream/master
```

**Comment by Johann-S** [comment]
I think it's ok now @XhmikosR

**Comment by XhmikosR** [comment]
@Johann-S 

1. this PR doesn't add sourcemaps for the non minified builds
2. you have actually deleted the minified files

**Comment by Johann-S** [comment]
But the goal of a source map it's to allow people to read a minified source.
So I don't understand why do you want to add source map for non minified file.

I'll add minified file to my PR sorry.

On Tue, Sep 08, 2015 at 6:20 pm, XhmikosR < notifications@github.com [notifications@github.com] > wrote:
@Johann-S [https://github.com/Johann-S]

 1. this PR doesn't add sourcemaps for the non minified builds
 2. you have actually deleted the minified files

—
Reply to this email directly or view it on GitHub
[https://github.com/twbs/ratchet/pull/787#issuecomment-138618191] .[https://github.com/notifications/beacon/ABnIlvtHHahEuUmEXBhHnM69CgeD-j1Dks5ovwJfgaJpZM4E9kAm.gif]

**Comment by XhmikosR** [comment]
@Johann-S: can you rebase and add back the minified files and the map files too?

**Comment by Johann-S** [comment]
Done @XhmikosR :+1:

**Comment by XhmikosR** [comment]
@Johann-S: still needs work. It shouldn't have the changes in dist/js/ratchet.js and it should be one commit.

**Comment by Johann-S** [comment]
Yes I'll squash my commits

File: dist/js/ratchet.js [hunk 1]
```diff
@@ -113,8 +113,8 @@
       }
       modal.dispatchEvent(eventToDispatch);
       modal.classList.toggle('active');
+      event.preventDefault(); // prevents rewriting url (apps can still use hash values in url)
     }
-    event.preventDefault(); // prevents rewriting url (apps can still use hash values in url)
```
Comment (Johann-S) [inline dist/js/ratchet.js]: It's strange I rebased. Do you know @XhmikosR how can I revert this change ?

Comment (XhmikosR) [inline dist/js/ratchet.js]: You need to rebase again, change that, commit, continue and then rebase when done and squash to one.

Comment (Johann-S) [inline dist/js/ratchet.js]: I cannot rebase from ``upstream master`` because my current branch is up to date (I did : ``git pull --rebase upstream master``)

Comment (XhmikosR) [inline dist/js/ratchet.js]: You can still do `git rebase -i upstream/master`.

**Comment by XhmikosR** [comment]
@Johann-S: can you do one last rebase and also update the patch for grunt-sass if needed?

**Comment by Johann-S** [comment]
Done :+1:

**Comment by XhmikosR** [comment]
@Johann-S: looks good, just a couple final thoughts.

1. Maybe we should enable sourcemaps for docs.css too
2. Might be wise to do what we do in BS 4 for *.map files  
```
# Don't diff or textually merge source maps
*.map   binary
```

**Comment by Johann-S** [comment]
All it's done @XhmikosR :wink:

**Comment by XhmikosR** [comment]
Almost there, please run grunt and commit the docs map too.

**Comment by Johann-S** [comment]
If this build passe https://travis-ci.org/twbs/ratchet/builds/99337086 I think it will be ok :pray:

**Comment by XhmikosR** [comment]
Thanks!

**Comment by Johann-S** [comment]
Thank you for your feedbacks and patience ! :smile:
